### PR TITLE
:recycle: Use `infinity` as a tombstone value for floating-point types

### PR DESCRIPTION
**Pull request by elbeno**

### CODE REVIEW

**Comment by lukevalenty** [comment]
I'm not 100% convinced we should have a default sentinel value for floating point. Someone might also choose `NaN` or even a regular number that they know would never be valid in their use case.

**Comment by lukevalenty** [comment]
Ok, in a side comment @elbeno mentioned that `NaN` is not comparable, so it can't be used as a sentinel.

File: include/stdx/optional.hpp [hunk 1]
```diff
@@ -186,7 +194,9 @@ template <typename T, typename TS = tombstone_traits<T>> class optional {
 
     [[nodiscard]] friend constexpr auto operator<(optional const &lhs,
                                                   optional const &rhs) -> bool {
-        return lhs.val < rhs.val and rhs.has_value();
+        return lhs.has_value() and rhs.has_value()
+                   ? lhs.val < rhs.val
+                   : not lhs.has_value() and rhs.has_value();
```
Comment (lukevalenty) [inline include/stdx/optional.hpp:199]: question: an optional without a value is always considered less than an optional with any value?

**Review by lukevalenty — APPROVED** [review]

File: include/stdx/optional.hpp (hunk 1 above)
Comment (elbeno) [reply include/stdx/optional.hpp:199]: that's right.

**Pull request merged**